stdlib/runtime/numbers.gr

```diff
@@ -2465,11 +2030,22 @@ provide let coerceNumberToFloat64 = (x: Number) => {
   WasmI32.toGrain(result): Float64
 }
 
+@unsafe
+let _SIMPLE_NUMBER_MAX = 0x3FFFFFFFn
+@unsafe
+let _SIMPLE_NUMBER_MIN = 0xC0000001n
+
 @unsafe
 provide let coerceInt32ToNumber = (x: Int32) => {
-  WasmI32.toGrain(
-    reducedInteger(WasmI64.extendI32S(boxedInt32Number(WasmI32.fromGrain(x))))
-  ): Number
+  let x = WasmI32.load(WasmI32.fromGrain(x), 4n)
+  let result = if (
+    WasmI32.gtS(x, _SIMPLE_NUMBER_MAX) || WasmI32.ltS(x, _SIMPLE_NUMBER_MIN)
+  ) {
+    newInt64(WasmI64.extendI32S(x))
+  } else {
+    tagSimple(x)
+  }
```

Comment:
This mostly duplicates the logic of `reducedInteger`. Makes me feel like we should just use that with the extend.

Reply:
Good point; I think that would also remove the need for the simple number constants

Reply:
Actually now that you bring this up I think there may be a bug with the "minimum" values; `_SMIN32_I64` is defined as `-0x7FFFFFFFN` but it actually should be one lower than that (since twos complement signed ints can hold one more negative value than positive)

Reply:
That's true. And explains why I thought the simple min and max were tagged, since simple min should be `0xC0000000`. Good catch! And something hard to catch in tests.